src/lib/types/src/Flow/Types/Type/Native/String/StringTypeChecker.php

```diff
@@ -108,6 +108,35 @@ public function isFloat() : bool
         return \is_numeric($this->string) && \str_contains($this->string, '.');
     }
 
+    public function isHTML() : bool
+    {
+        if ($this->string === '') {
+            return false;
+        }
+
+        if ('<' !== $this->string[0]) {
+            return false;
+        }
+
+        if (\preg_match('/(<!doctype(.+?)>)?<html(.+?)>(.+?)<\/html>/im', $this->string) === 1) {
```

Comment:
I think I would first check if HTMLDocument is available and try to use it, and when it fails then fallback to DOMDocument with this Regexp.